smart_open/tests/test_smart_open.py

```diff
@@ -890,7 +902,58 @@ def test_write_read_bz2(self):
         """Can write and read bz2?"""
         test_file = tempfile.NamedTemporaryFile('wb', suffix='.bz2', delete=False).name
         self.write_read_assertion(test_file)
-
+    
+    '''
+    # TO-DO: fix fd-leak for local gz,bz2 on python 2.6.9
+    def test_close_gz(self):
+        """Is gzip closed correctly?"""
+        test_file = tempfile.NamedTemporaryFile('wb', suffix='.gz', delete=False).name
+        self.close_assert(test_file)
+    
+    def test_close_bz2(self):
+        """Is bz2 closed correctly?"""
+        test_file = tempfile.NamedTemporaryFile('wb', suffix='.bz2', delete=False).name
+        self.close_assert(test_file)
+    '''
+    
+    @responses.activate
+    def test_read_write_http_gz(self):
+        """Does modes correctly for gz over http?"""
+        text= 'line1\nline2'
```

Comment:
this is not gzipped content. Please test with gzipped content. There are examples of archived files in https://github.com/RaRe-Technologies/smart_open/tree/master/smart_open/tests/test_data